tests/unit/CoreParagraphTest.php

```diff
@@ -0,0 +1,387 @@
+		$this->assertNotEmpty( $block['clientId'], 'The clientId should be present' );
+
+		$this->assertEmpty( $block['innerBlocks'], 'There should be no inner blocks' );
+		$this->assertEquals( 'core/paragraph', $block['name'], 'The block name should be core/paragraph' );
+		$this->assertEmpty( $block['parentClientId'], 'There should be no parentClientId' );
+		$this->assertNotEmpty( $block['renderedHtml'], 'The renderedHtml should be present' );
+
+		$attributes = $block['attributes'];
+		$this->assertEquals(
+			[
+				'align'           => 'center',
+				'anchor'          => null,
+				'backgroundColor' => 'pale-cyan-blue',
+				'className'       => null,
+				'content'         => 'This is a test paragraph with various attributes.',
+				'cssClassName'    => 'has-text-align-center has-vivid-red-color has-pale-cyan-blue-background-color has-text-color has-background has-large-font-size has-helvetica-arial-font-family',
+				'dropCap'         => false,
+				'direction'       => null,
+				'fontFamily'      => 'helvetica-arial',
+				'fontSize'        => 'large',
+				'gradient'        => null,
+				'lock'            => null,
+				'metadata'        => null,
+				'placeholder'     => null,
+				'style'           => null,
+				'textColor'       => 'vivid-red',
+			],
+			$attributes
+		);
+	}
+
+	/**
+	 * Test retrieval of core/paragraph block with drop cap and custom styles.
+	 *
+	 * Attributes covered:
+	 * - dropCap
+	 * - style (typography and spacing)
+	 * - content
+	 * - cssClassName
+	 *
+	 * @return void
+	 */
+	public function test_retrieve_core_paragraph_with_drop_cap_and_custom_styles() {
+		$block_content = '
+			<!-- wp:paragraph {"dropCap":true,"style":{"typography":{"lineHeight":"2","textTransform":"uppercase"},"spacing":{"padding":{"top":"20px","right":"20px","bottom":"20px","left":"20px"}}}} -->
+			<p class="has-drop-cap" style="padding-top:20px;padding-right:20px;padding-bottom:20px;padding-left:20px;line-height:2;text-transform:uppercase">This is a paragraph with drop cap and custom styles.</p>
+			<!-- /wp:paragraph -->
+		';
+
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
+
+		$this->assertEquals( 1, count( $actual['data']['post']['editorBlocks'] ) );
+
+		$block = $actual['data']['post']['editorBlocks'][0];
+
+		$this->assertEquals( 'core/paragraph', $block['name'], 'The block name should be core/paragraph' );
+
+		$attributes = $block['attributes'];
+		$this->assertEquals(
```

Comment:
Leave a `// previously untested` by the new attribute values.

Reply:
Added.